### PR TITLE
#8578 SEARCH_WITH_FILTER action: add popup support

### PR DESCRIPTION
**Pull request by belom88**
## Description
Add popup support to `searchOnStartEpic`

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [X] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#8578

**What is the new behavior?**
Works as expected

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information

Steps to reproduce on local environment:
* Set `"showInMapPopup": true` in the config https://github.com/geosolutions-it/MapStore2/blob/0e3b035dc1a3928dac19b8dbf5ba45330cdd3a29/web/client/configs/localConfig.json#L335
* Create a map with "Manhattan (NY) points of interest";
* Open new map with URL: `http://localhost:8081/#/viewer/openlayers/19?actions=[{%22type%22:%22SEARCH:SEARCH_WITH_FILTER%22,%22cql_filter%22:%22NAME='art'%22,%22layer%22:%22tiger:poi2%22}]`

### CODE REVIEW

File: web/client/epics/search.js [hunk 1]
```diff
@@ -321,9 +324,14 @@ export const searchOnStartEpic = (action$, store) =>
                     .switchMap(({ type, geometry, typeName }) => {
                         let coord = pointOnSurface({ type, geometry }).geometry.coordinates;
                         const latlng = {lng: coord[0], lat: coord[1] };
+                        const {x, y} = reproject(coord, 'EPSG:4326', 'EPSG:3857');
 
                         if (coord) { // trigger get feature info
-                            return Rx.Observable.of(featureInfoClick({latlng}, typeName, [typeName], {[typeName]: {cql_filter: cqlFilter}}), showMapinfoMarker());
+                            return Rx.Observable.of(featureInfoClick({latlng}, typeName, [typeName], {[typeName]: {cql_filter: cqlFilter}}), showMapinfoMarker())
```
Comment (allyoucanmap) [inline web/client/epics/search.js:330]: There is a marker on the map after closing the popup and this is due to the showMapinfoMarker() action.
We should avoid to dispatch showMapinfoMarker if the isMapPopup is true bacause the marker should be hidden, the popup is already showing the position.

https://user-images.githubusercontent.com/19175505/194865753-0830262a-05fd-48c1-8a51-a907e1a02234.mp4

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/epics/search.js (hunk 1 above)
Comment (jusabatier) [reply web/client/epics/search.js]: I also think it's better to hide the marker when we use popup display option.

File: web/client/epics/search.js [hunk 2]
```diff
@@ -299,7 +303,7 @@ export const zoomAndAddPointEpic = (action$, store) =>
 export const searchOnStartEpic = (action$, store) =>
     action$.ofType(SEARCH_LAYER_WITH_FILTER)
         .switchMap(({layer: name, "cql_filter": cqlFilter}) => {
-            const state = store.getState();
+            const state = store.getState(store.getState());
```
Comment (allyoucanmap) [inline web/client/epics/search.js:306]: please fix this get state

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test on dev and let us know when we can backport this on branch stable 2022.02.xx, thanks

**Comment by ElenaGallo** [comment]
Test passed on DEV, @belom88 please backport this on branch stable 2022.02.xx. Thanks

**Pull request merged**